back/app/services/participation_permissions_service.rb

```diff
@@ -329,11 +338,11 @@ def disliking_limit_reached?(phase, user)
     phase.reacting_dislike_limited? && user.reactions.down.where(reactable: phase.ideas).size >= phase.reacting_dislike_limited_max
   end
 
-  def permission_denied_reason(user, action, phase)
-    permissions_service.denied_reason_for_resource user, action, phase
-  end
-
-  def permissions_service
-    @permissions_service ||= ::PermissionsService.new
+  def project_visible_disabled_reason(project, user)
+    if project.visible_to == 'admins' && !user.admin?
+      POSTING_DISABLED_REASONS[:project_only_visible_to_admins]
+    elsif project.visible_to == 'groups' && !user.in_any_groups?(project.groups)
+      POSTING_DISABLED_REASONS[:project_only_visible_to_groups]
+    end
```

Comment:
This is the method added to check if the project is visible to the user or not. Wasn't sure whether to put it here or in the permissions service, but probably just another reason why we need to refactor all of this to make it more coherent.